### ISSUE TITLE
Why missing quotes around `$()` is not detected?

### ISSUE THREAD

**Issue by tbhaxor** [koalaman/shellcheck#2813]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): `online`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2046)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

I am not sure if this comes under exception or not? Please confirm this.

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

root_dir=$(realpath "$(dirname "${BASH_SOURCE[0]}")/../")
echo "$root_dir"
```

#### Here's what shellcheck currently says:

_Nothing_

#### Here's what I wanted or expected to see:

I want it to detect and throw warning

**Comment by koalaman** [comment]
Assignments like `var=$(..)` are not subject to word splitting or pathname expansion, and therefore don't need quotes. Quotes are allowed, but have no effect.
